networkit/csbridge.py

```diff
@@ -0,0 +1,141 @@
+# local imports
+from .support import MissingDependencyError
+from .structures import Partition
+
+# external imports
+try:
+	import ipycytoscape
+except ImportError:
+	have_cyto = False
+else:
+	have_cyto = True
+
+try:
+	import seaborn
+except ImportError:
+	have_seaborn = False
+else:
+	have_seaborn = True
+
+def widget_from_graph(G, node_scores = None, node_partition = None, node_colors = None, show_ids = True):
+	""" 
+	Creates a ipycytoscape-widget from a given graph. The returned widget already contains
+	all nodes and edges from the graph. The graph is colored using an array of norm. rgb-values
+	based on seaborn perceptually uniform color map (rocket) or a user given custom color array. 
+	See matplotlib color maps for the correct formatting: 
+	https://matplotlib.org/api/_as_gen/matplotlib.colors.Colormap.html#matplotlib.colors.Colormap 
+
+	Parameters:
+	-----------
+	G : networkit.graph.Graph
+		Graph (nodes, edges), which should be visualized
+	node_scores : list of numbers
+		List of scores for each nodes, for example scores from a centrality measure. This is 
+		used for color-calculation of the nodes (continuous distribution). Provide either 
+		node_scores or node_partition - not both.
+	node_partition : networkit.structures.Partition
+		Partition object. This is used for color-calculation of the nodes (discrete distribution). 
+		Provide either node_scores or node_partition - not both. 	
+	node_colors : list of tuples
+		Array consisting of normalized rgb-values. If none is given, seaborn.color_palette.colors is used
+	show_ids : 	boolean	
+		Set whether node ids should be visible in plot-widget. Is set to True by default. 
+	"""
+	# Sanity checks
+	if not have_cyto:
+		raise MissingDependencyError("ipycytoscape")
+
+	if node_scores is not None and node_partition is not None:
+		raise InputError("Provide either node_scores or node_partition - not both")
+
+	# Color calculation: score = continuous distribution, partition = discrete distribution
+	hc_colors = []
+
+	# Partition
+	if node_partition is not None:
+		if node_colors is not None:
+			palette = node_colors
+		else:
+			if not have_seaborn:
+				raise MissingDependencyError("seaborn")
+			palette = seaborn.color_palette("hls", node_partition.numberOfSubsets())
+
+		partitions = node_partition.getVector()
+
+		for i in G.iterNodes():
+			hc_colors.append((palette[partitions[i]][0] * 255, palette[partitions[i]][1] * 255, palette[partitions[i]][2] * 255))
```

Comment:
What if node_colors does not have 256 colors?

Reply:
In line 61 the palette is initialized with the number of partitions. This does scale the number of colors in the palette. I add a check in any case.

Reply:
Yes, but I can supply a `node_colors` array to bypass the palette generation, right? Shouldn't we dynamically adapt this to the number of elements in `node_colors`?

Reply:
I would say we add another check, in case a user provides a color and it does not contain enough colors to match the number of partitions. Generating more colors (especially with seaborn) is easy - manipulating and extrapolating new colors in a given set maybe not that much :(

Reply:
Maybe it is enough to replace to scaling (0 - 255) to whatever the number of colors in `node_colors` is? Can we easily get the number of colors in a seaborn palette?